eduvpn/ui/stats.py

```diff
@@ -0,0 +1,139 @@
+import os
+import logging
+import time
+from datetime import timedelta
+
+from ..nm import get_iface, get_ipv4, get_ipv6, get_timestamp
+
+logger = logging.getLogger(__name__)
+
+LINUX_STATS_FILE = "/sys/class/net/{iface}/statistics/{filename}"
+
+
+class NetworkStats():
+    def __init__(self, interval):
+        self.iface = None
+        self.interval = interval
+        self.addresses = {}
+        self.files = {}
+        self.start_bytes = {}
+        self.timestamp = None
+        self.default_text = "N/A"
+
+    def open_file(self, filename):
+        """
+        Helper function to open a statistics network file
+        """
+        filepath = LINUX_STATS_FILE.format(iface=self.iface, filename=filename)
+        if not os.path.isfile(filepath):
+            logger.info(f"Network Stats: {filepath} is not a file")
+            return None
+        return open(filepath, "r")
+
+    def get_human_readable_bytes(self, total_bytes):
+        """
+        Helper function to calculate the human readable bytes.
+        E.g. B, KB, MB, GB, TB.
+        """
+        hr_bytes = total_bytes
+        suffix = ""
+        for suffix in ["B", "KB", "MB", "GB", "TB"]:
+            if hr_bytes < 1024.0:
+                break
+            if suffix != "TB":
+                hr_bytes /= 1024.0
+
+        return f"{hr_bytes:.2f} {suffix}"
+
+    def get_file_bytes(self, filename, typename):
+        """
+        Helper function to get a statistics file to calculate the total data transfer
+        """
+        # If interface wasn't set
+        if self.iface is None:
+            self.iface = get_iface()
+
+        # Try to get it again
+        if self.iface is None:
+            logger.debug("Network Stats: failed to get iface")
+            return self.default_text
+
+        # Open the file if it is not open yet
+        current_file = self.files.get(typename, None)
+        if current_file is None:
+            current_file = self.open_file(filename)
+            if current_file is None:
+                return self.default_text
+            self.files[typename] = current_file
+
+        # Get the statistic from the file
+        # and go to the beginning
+        stat = float(current_file.readline())
```

Comment:
i think we should move the opening and parsing of the file to a separate helper function, this method is getting quite long.

Reply:
makes sense!